yarn/assets/configuration/spec.yaml

```diff
@@ -95,14 +97,22 @@ files:
               type: boolean
           - name: collect_apps_all_states
             description: |
-              Enable this parameter when using `collect_app_metrics` to collect metrics from applications of all states.
+              Enable this parameter to collect metrics from applications of all states when collecting yarn.app metrics.
               For example, collect metrics from applications with 'NEW', 'SUBMITTED', 'FINISHED', and other states.
+
+              This instance-level option of `collect_apps_all_states` overrides the init_config-level
+              `collect_apps_all_states` option.
             value:
               example: false
               type: boolean
           - name: collect_apps_states_list
             description: |
-              Configure the application states of which to collect application metrics.
+              Configure the application states of which to collect application metrics. 
+              Valid options are 'ALL', 'NEW', 'NEW_SAVING', 'SUBMITTED', 'ACCEPTED',
+              'RUNNING', 'FINISHED', 'FAILED', and 'KILLED'.
+
+              This instance-level option of `collect_apps_states_list` overrides the init_config-level
+              `collect_apps_states_list` option and also the instance-level `collect_apps_all_states`.
```

Comment:
Shouldn't `collect_apps_all_states` override the init_config and instance level `collect_apps_states_list`? 

> init_config.collect_apps_states_list < init_config.collect_apps_all_states < instance.collect_apps_states_list < instance.collect_apps_all_states